pubsub/streaming-analytics/README.md

```diff
@@ -0,0 +1,150 @@
+     > If you are developing a production app, specify more granular permissions than **Project > Owner**.
+     > For more information, see [Granting roles to service accounts].
+
+   * Click **Create**. A JSON file that contains your key downloads to your computer.
+
+1. Set your `GOOGLE_APPLICATION_CREDENTIALS` environment variable to point to your service account key file.
+
+   ```bash
+   export GOOGLE_APPLICATION_CREDENTIALS=path/to/your/credentials.json
+   ```
+
+1. Create a Cloud Storage bucket. Grant the Default Compute Engine Service Account Object Admin Access to your bucket.
+
+   ```bash
+   gsutil mb gs://your-gcs-bucket
+   
+   BUCKET=gs://your-gcs-bucket
+   PROJECT=${gcloud config get-value project}
+   PROJECT_NUMBER=$(gcloud projects describe ${PROJECT} --format="value(projectNumber)")
+   
+   gsutil iam ch \
+     serviceAccount:${PROJECT_NUMBER}-compute@developer.gserviceaccount.com:objectAdmin \
+     ${BUCKET}
+   ```
+   
+ 1. Start a Cloud Scheduler job that publishes one message to a Cloud Pub/Sub topic every minute.
+ 
+    ```bash
+    gcloud pubsub topics create cron-topic
+    
+    gcloud scheduler jobs create pubsub my-job --schedule="* * * * *" \
+      --topic=cron-topic --message-body="Hello Pub/Sub!"
+    ```
+
+## Setup
+
+The following instructions will help you prepare your development environment.
+
+1. Download and install the [Java Development Kit (JDK)].
+   Verify that the [JAVA_HOME] environment variable is set and points to your JDK installation.
+
+1. Download and install [Apache Maven] by following the [Maven installation guide] for your specific operating system.
+
+1. Clone the `java-docs-samples` repository.
+
+    ```bash
+    git clone https://github.com/GoogleCloudPlatform/java-docs-samples.git
+    ```
+
+1. Navigate to the sample code directory.
+
+   ```bash
+   cd java-docs-samples/pubsub/streaming-analytics
+   ```
+
+## Streaming Analytics
+
+### Google Cloud Pub/Sub to Google Cloud Storage
+
+* [PubSubToGCS.java](src/main/java/com/examples/pubsub/streaming/PubSubToGCS.java)
+
+To run the example and write output files in the desired Cloud Storage location.
+
+```bash
+mvn compile exec:java \
+  -Dexec.mainClass=com.examples.pubsub.streaming.PubSubToGCS \
+  -Dexec.cleanupDaemonThreads=false \
+  -Dexec.args="\
+    --windowSize=2 \
+    --inputTopic=projects/${PROJECT}/topics/cron-topic \
+    --output=gs://${BUCKET}/output- \
+    --runner=DataflowRunner"
+```
+
+After the job has been submitted, you can check its status in the [GCP Console Dataflow page].
+
+## Cleanup
+
+1. Delete the Cloud Schedule job. 
+    ```bash
+    gcloud scheduler jobs delete my-job
+    ```
+
+1. `Ctrl + C` to stop the program in your terminal.
+
+1. Cancel the DataFlow job in [GCP Console Dataflow page]. Stop the pipeline without draining it. 
+
+1. Delete the topic. 
+   ```bash
+   gcloud pubsub topics delete cron-topic
```

Comment:
You also should delete the subscription(s).

```
# Delete all the subscriptions associated to the topic
gcloud pubsub topics list-subscriptions cron-topic | grep $PROJECT | xargs gcloud pubsub subscriptions delete
```

You can probably find a simpler way to do it.

Reply:
Luckily when we cancel the Dataflow job, it deletes this subscription. :)